include/mbedtls/platform.h

```diff
@@ -58,7 +58,7 @@ extern "C" {
 #endif
 #endif
 #if !defined(MBEDTLS_PLATFORM_STD_VSNPRINTF)
-#if _WIN32_WINNT < 0x0A00 /* < Windows 10 */
+#if _WIN32 && _MSC_VER < 1900 /* Before MSVC 14.0 / Visual Studio 2015 */
```

Comment:
I'm not sure this is correct either. What if the platform is windows and but compiler is something other than MSVC? I think you need to check if `_MSC_VER` is defined before checking its value.

Also, to check if macro is defined, please use `defined(MACRO)` rather than just `MACRO`.

Finally, since this check is repeated a few times, perhaps we should defined a macro for it, eg `WIN32_WITH_SNPRINTF_FAMILY`

Reply:
Aggreed.

I can see that MSVC has got a conformance update in the Visual Studio 2015. It's claimed to fully support the C99 standard library except for features requiring missing compiler functionality. This makes it an outstanding release. I'd turn that condition into a compile time define pointing to the VS version rather than particular function family as the compliance changes have most probably been wider that just the niche printing functions.